modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/OrchestratorService.java

```diff
@@ -147,23 +161,41 @@ public void updateManagedService(String managedServiceName, Ocl ocl) {
     public void startManagedService(String managedServiceName) {
         if (!this.databaseOrchestratorStorage.isExists(managedServiceName)) {
             throw new EntityNotFoundException(
-                    "Managed service " + managedServiceName + " not found");
+                "Managed service " + managedServiceName + " not found");
         }
+        int inBuildingCount = 0;
         for (OrchestratorPlugin orchestratorPlugin : plugins) {
             ServiceStatusEntity serviceStatusEntity =
-                    this.databaseOrchestratorStorage.getServiceDetailsByNameAndPlugin(
-                            managedServiceName,
-                            orchestratorPlugin);
+                this.databaseOrchestratorStorage.getServiceDetailsByNameAndPlugin(
+                    managedServiceName,
+                    orchestratorPlugin);
+            if (Objects.nonNull(serviceStatusEntity.getServiceState())) {
+                if (serviceStatusEntity.getServiceState().equals(ServiceState.STARTED)
+                    || serviceStatusEntity.getServiceState().equals(ServiceState.STARTING)) {
+                    log.warn("Managed service:{} is starting or started in plugin:{}",
+                        managedServiceName,
+                        orchestratorPlugin.getClass().getSimpleName());
+                    inBuildingCount++;
+                    if (inBuildingCount >= plugins.size()) {
+                        throw new RuntimeException(
+                            "Managed service " + managedServiceName
+                                + " is starting or already started");
+                    }
+                    continue;
+                }
+            }
             try {
                 serviceStatusEntity.setServiceState(ServiceState.STARTING);
                 this.databaseOrchestratorStorage.store(serviceStatusEntity);
-                orchestratorPlugin.startManagedService(managedServiceName);
-                serviceStatusEntity.setServiceState(ServiceState.STARTED);
-                this.databaseOrchestratorStorage.store(serviceStatusEntity);
-
+                executorService.execute(
```

Comment:
We need async to be on all services.. register, start, stop, unregister. 
With our current plugins, we have only "start" doing something. But in our end goal, all these 4 methods will have actions which will take longer.

Reply:
Ok. I add the API of unregister and async register, start, stop, unregister services